rfd/0081/health-check-draft.md

```diff
@@ -0,0 +1,58 @@
+# Mariposa Health Check Service
+
+The Mariposa (RFD36) health check service will be responsible for monitoring the health of user services. It establishes a set of supported health checks and provides an API for defining health checks that should be ran against project services. These health check definitions will be persisted in a proper data store (Postgres/Manatee?).
+
+The Health Check service process(es) will read user health check definitions, and coordinate which health check process will be responsible for running any given definition. 
+
+Failed health checks will be broadcast to interested services, such as the Convergence service.
+
+## Health check types
+
+The available health checks include:
+
+* HTTP(s) - Test the response of an HTTP request against a given service/port. Any part of the response may be validated including response code, headers, or body.
+* TCP - Test that a service/port is accepting TCP connections.
+* ContainerPilot - Interface with ContainerPilot running within a service container and accept the health state returned by it.
+
+In addition to the health checks specified, the health check service will interface with VMAPI and ChangeFeed to understand when a container state transitions between stopped/paused/running. Stopped/paused will be considered unhealthy and the above health checks will not be attempted against containers in these states.
+
+## Health check configuration
+
+Health check definitions files will be YAML, to be consistent with project manifests and metadata files. 
+
+All health check definitions will include:
+
+* Healthy poll interval - How often the service is checked while deemed healthy
+* Unhealthy poll interval - Defaults to healthy poll interval, but may be desirable to have a smaller value here for two reasons: faster recovery detection, and faster detection of initial healthiness, there could be a back-off associated with this
+* Timeout - How long to wait for the health check before aborting and considering the service unhealthy
+* Recheck - A number of times to re-check a failed health check before transitioning the service to an unhealthy state, default to zero.
+
+## Endpoints
```

Comment:
The endpoint is for debugging convenience, yes? Since any changes in health would be broadcast to the Convergence service via changefeed, no?